### ISSUE TITLE
Chai-1 for proteins >2048 AAs

### ISSUE THREAD

**Issue by tommyhuangthu** [chaidiscovery/chai-lab#77]
Dear Chai Team,

Thank you very much for releasing the powerful Chai-1 platform. Our lab is very interested in a few very large proteins that are much longer than 2048 AAs. One example is the human Usherin (UniProt ID: O75445) that has 5202 AAs. I wonder if your team plans to release a version for modeling such ultra-large proteins. And if so, any suggestions on the required resources for launching the program in a HPC cluster?

Best regards,

**Comment by MattMcPartlon** [comment]
Hi Tommy,

This isn't something we're actively working on now but it is something we hope to support in future releases. Unfortunately memory/time scaling in the current architecture is essentially cubic in sequence length 😞 . There are some engineering tricks that could get us to support ~5k length range but inference would require multiple GPUs and would still be extremely slow.

TLDR: hoping to support at some point in the future but no current plans.

**Comment by tommyhuangthu** [comment]
> Hi Tommy,
> 
> This isn't something we're actively working on now but it is something we hope to support in future releases. Unfortunately memory/time scaling in the current architecture is essentially cubic in sequence length 😞 . There are some engineering tricks that could get us to support ~5k length range but inference would require multiple GPUs and would still be extremely slow.
> 
> TLDR: hoping to support at some point in the future but no current plans.

Thank you for this information.
